lib/utils.pm

```diff
@@ -328,80 +328,45 @@ sub workaround_type_encrypted_passphrase {
 # screen lock is necessary when switch back to x11
 # returns without login if any of the tags in param matches
 sub ensure_unlocked_desktop {
-    my ($extra_tags) = @_;
-    $extra_tags //= [];
-
-    # try to stop screenlocker / blanking
-    send_key "esc";
-
-    # prevent the still running screen lock from consuming next key press
-    wait_still_screen(1);
-
-    my $tags = [];
-    # wait for run prompt or screen locker
-    push @$tags, @$extra_tags;
-    push @$tags, 'generic-desktop';
-    push @$tags, 'screenlock';
-    push @$tags, 'gnome-screenlock-password';
-    assert_screen($tags);
-
-    for my $tag (@$extra_tags) {
-        if (match_has_tag $tag) {
-            # caller provided tag matched
-            # caller needs to deal with it
-            return;
-        }
-    }
-
-    # no screenlocker detected
-    if (match_has_tag 'generic-desktop') {
-        if (check_var('DESKTOP', 'gnome')) {
-            # gnome might show the old 'generic desktop' screen although that is
-            # just a left over in the framebuffer but actually the screen is
-            # already locked so we have to try something else to check
-            # responsiveness.
-            # open run command prompt (if screen isn't locked)
-            send_key "alt-f2";
-            if (check_screen([qw/desktop-runner/])) {
-                send_key "esc";
-                assert_screen([qw/generic-desktop/]);
-                return;
-            }
-            else {
-                assert_screen([qw/gnome-screenlock-password screenlock/]);
+    my $counter = 10;
+    while ($counter--) {
```

Comment:
This seems to be the main trick, brute force retry. The previous behaviour seemed to be more correct as it explicitly tries to cover all cases, of course, it can fail so there is still something that we missed. I would not mind the multiple retry approach but this should be properly commented.

Some potential corner cases:
* what if the while loop does not reach "last" within 10 tries? Shouldn't you die if you exited the while loop and $counter is zero?
* what if the displaymanager or generic desktop changes and the behaviour of the SUT is correct but the needle just does not match, would it show the locked screen at last and then you can not create the needle? You could test this by removing the generic-desktop and/or displaymanager needles, let it fail (because no matching needles exist) and then check if the failure screenshots give you a good hint what is actually the case *and* that one could create an updated needle from the failure screenshot